rfcs/021-logging/README.md

```diff
@@ -0,0 +1,113 @@
+
+There is also proliferation of `logstash-transit` services in the platform adding cost and complexity.
+
+## Proposed solution
+
+If we can take advantage of the container level logging drivers available in ECS we can return to a position where our applications have no configuration for transporting logs elsewhere. This would eliminate some of the current configuration and infrastructure complexity.
+
+At time of writing the logging drivers available for the [AWS Fargate](https://aws.amazon.com/fargate/) containers that make up the vast majority of our services [are limited to `awslogs`, `splunk`, and `awsfirelens`](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LogConfiguration.html). 
+
+### Firelens
+
+`awsfirelens` is a solution for custom log routing that makes use of a [Fluent Bit](https://fluentbit.io/) or [Fluentd](https://www.fluentd.org/) container as a [sidecar](https://www.oreilly.com/library/view/designing-distributed-systems/9781491983638/ch02.html) to your application container.
+
+AWS provides pre-built images for running both Fluent Bit and Fluentd in ECS and provide [instructions for setting up custom log routing](https://docs.aws.amazon.com/AmazonECS/latest/userguide/using_firelens.html).
+
+Firelens requires you to add a container to your task definition, though that could be provided with fluent bit configuration from the existing [`terraform-aws-ecs-service`](https://github.com/wellcomecollection/terraform-aws-ecs-service) module allowing us to standardise that configuration.
+
+We may still need to provide a mechanism to forward logs collected to Elasticsearch. 
+
+### Fluent Bit
+
+AWS recommends using Fluent Bit as a log collector over Fluentd [due to performance and resource considerations](https://aws.amazon.com/blogs/opensource/centralized-container-logging-fluent-bit/). 
+
+Fluent Bit is similar to Logstash in function and is capable of [relaying logs to Logstash](https://github.com/fluent/fluent-bit-tutorials/blob/master/ship_to/logstash.md) if required. See the [Fluent Bit](https://docs.fluentbit.io/manual/) manual for details of input & output plugins.
+
+### Routing logs to Elasticsearch
+
+At present we route logs through a cluster-local Logstash service that is configured to authenticate with a Elastic Cloud hosted Elasticsearch instance and forward logs there with several `logstash-transit` services across the estate.
+
+There are a few options for routing to Elasticsearch.
+
+- Directly to Elasticsearch from the Fluent Bit sidecar.
+  - (pro) No other log forwarding infrastructure.
+  - (con) Elasticsearch config would be required in each service.
+- Forwarding logs to a cluster-local collector as we do now.
+  - (???) We can use the Fluent Bit [`forward`](https://docs.fluentbit.io/manual/input/forward) input and so replace Logstash with Fluent Bit which _might_ be more reliable.
+  - (con) Multiple `???-transit` services would still be required.
```

Comment:
This is still the case, but just mode hidden though?

Reply:
We still need to route logs from there to ElasticSearch. What we are discussing in this section is how to do that. Lots of collectors (logstash-transit services) or one collector service made available via PrivateLink.